File: algorithm_introduction/ss26_maximum_flow/cpp/hello.cpp
#include <stdio.h>
#include <vector>

#define N 3

using namespace std;

struct Edge {
  int u;
  Edge() {}
  Edge(int u) : u(u) {}
};

Edge E[N];

void init(Edge E[]) { E[1] = Edge(100); }

int main() {
  init(E);
  printf("%d\n", E[1].u);
}

File: algorithm_introduction/ss26_maximum_flow/python/hello.py
class Edge:
    def __init__(self, u):
        self.u = u


E = [Edge(i) for i in range(3)]

_E = []
for e in E:
    _E.append(e)

_E[0].u = 100

print(E[0].u)
print(_E[0].u)
